python/tank/authentication/login_dialog.py

```diff
@@ -39,6 +39,47 @@
 USER_INPUT_DELAY_BEFORE_SSO_CHECK = 300
 
 
+class QuerySiteAndUpdateUITask(QtCore.QThread):
+    """
+    This class uses a different thread to query if SSO is enabled or not.
+
+    We use a different thread due to the time the call can take, and
+    to avoid blocking the main GUI thread.
+    """
+
+    def __init__(self, parent):
+        """
+        Constructor.
+        """
+        QtCore.QThread.__init__(self, parent)
+        self._url_to_test = ""
+        self._sso_enabled = False
+
+    @property
+    def sso_enabled(self):
+        """Bool R/W property."""
+        return self._sso_enabled
+
+    @sso_enabled.setter
+    def sso_enabled(self, value):
+        self._sso_enabled = value
+
+    @property
+    def url_to_test(self):
+        """String R/W property."""
+        return self._url_to_test
+
+    @url_to_test.setter
+    def url_to_test(self, value):
+        self._url_to_test = value
+
+    def run(self):
+        """
+        Runs the thread.
+        """
+        self.sso_enabled = is_sso_enabled_on_site(self.url_to_test)
```

Comment:
The only thing the thread does is the SSO check (using the Shotgun Python API).
We only do a single call to the site. No worries if there are multiple concurrent calls.

The UI is updated by a slot call in the main thread once the 'run()' call is completed.